Find a functionally equivalent code:com.android.internal.policy.impl.PhoneWindowManager.selectAnimationLw:COMMENT
Method Modifier: public      internal    
Comment:/**
 * {@inheritDoc}
 */

@@ -2,11 +2,12 @@
     if (PRINT_ANIM)
         Log.i(TAG, "selectAnimation in " + win + ": transit=" + transit);
     if (win == mStatusBar) {
+        boolean isKeyguard = (win.getAttrs().privateFlags & PRIVATE_FLAG_KEYGUARD) != 0;
         if (transit == TRANSIT_EXIT || transit == TRANSIT_HIDE) {
-            return R.anim.dock_top_exit;
+            return isKeyguard ? -1 : R.anim.dock_top_exit;
         } else if (transit == TRANSIT_ENTER || transit == TRANSIT_SHOW) {
-            return R.anim.dock_top_enter;
+            return isKeyguard ? -1 : R.anim.dock_top_enter;
         }
     } else if (win == mNavigationBar) {
         // This can be on either the bottom or the right.

